app/uk/gov/hmrc/helptosavereminder/repo/HtsReminderRepository.scala

```diff
@@ -176,24 +176,49 @@ class HtsReminderMongoRepository @Inject() (mongo: MongoComponent)(implicit val
       logger.warn(s"nextSendDate for User: ${htsReminder.nino} cannot be updated.")
       Future.successful(false)
     } else {
+      val modifiedJson = if (htsReminder.endDate.nonEmpty) {
+        List(
+          Updates.set("optInStatus", htsReminder.optInStatus),
+          Updates.set("email", htsReminder.email),
+          Updates.set("firstName", htsReminder.firstName),
+          Updates.set("lastName", htsReminder.lastName),
+          Updates.set("daysToReceive", htsReminder.daysToReceive),
+          Updates.set("endDate", htsReminder.endDate.get)
+        )
+      } else {
+        List(
+          Updates.set("optInStatus", htsReminder.optInStatus),
```

Comment:
Why all of this repetition?

Reply:
the list in the else part can be extracted but the if else cannot be removed